### PR TITLE
Refactor lmi_dist and vllm to support best_of with RequestOutput

### PR DESCRIPTION
**Pull request by sindhuvahinis**
## Description ##

Brief description of what this PR is about

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: engines/python/setup/djl_python/tests/rolling_batch_test_scripts/test_rb_vllm_utils.py [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+
+import djl_python
+from djl_python.output_formatter import _json_output_formatter
+from djl_python.request import Request
+from djl_python.request_io import TextGenerationOutput, TextInput, Sequence, Token
+'''These Mock classes are in compliance with vllm RequestOutput version 0.4.2'''
+
+
+@dataclass
+class MockLogprob:
```
Comment (lanking520) [inline engines/python/setup/djl_python/tests/rolling_batch_test_scripts/test_rb_vllm_utils.py:20]: this mock class is good. You can also try to use the fake_rolling_batcher to do some of the work like this.

**Review by lanking520 — APPROVED** [review]

File: engines/python/setup/djl_python/tests/rolling_batch_test_scripts/test_rb_vllm_utils.py (hunk 1 above)
Comment (sindhuvahinis) [reply engines/python/setup/djl_python/tests/rolling_batch_test_scripts/test_rb_vllm_utils.py:20]: Yes, sure, we cam add this.

**Comment by sindhuvahinis** [comment]
Will run the LMI integration tests and then merge it, as it affects LMI, VLLM and NeuronVLLM.

**Comment by sindhuvahinis** [comment]
Ran integration tests 
https://github.com/deepjavalibrary/djl-serving/actions/runs/9407303778 - Gemma failed. But running it in my EC2 machine works though. 
Remaining models other than gemma also passed. https://github.com/deepjavalibrary/djl-serving/actions/runs/9409733042/job/25920234322
